### PR TITLE
Cleanup of metrics SDK

### PR DESCRIPTION
**Pull request by jack-berg**
While making some recent changes to the metrics SDK, I've noticed a handful of small things to cleanup, including:
- Typos
- Unused parameters on internal methods
- Redundant keywords
- Random other small things

### CODE REVIEW

File: sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/SdkDoubleCounter.java [hunk 1]
```diff
@@ -89,13 +89,6 @@ public void unbind() {
   static final class Builder extends AbstractInstrumentBuilder<SdkDoubleCounter.Builder>
       implements DoubleCounterBuilder {
 
-    Builder(
-        MeterProviderSharedState meterProviderSharedState,
-        MeterSharedState meterSharedState,
-        String name) {
-      this(meterProviderSharedState, meterSharedState, name, "", "1");
-    }
```
Comment (jack-berg) [inline sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/SdkDoubleCounter.java:97]: These builders are unused.

**Review by jkwatson — APPROVED** [review]
Thanks!

File: sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/state/WriteableMetricStorage.java [hunk 2]
```diff
@@ -7,6 +7,7 @@
 
 import io.opentelemetry.api.common.Attributes;
 import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.metrics.data.MetricData;
```
[the remark this answers is not in the document]
Comment (anuraaga) [reply sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/state/WriteableMetricStorage.java:10]: I think we usually fully qualify links that are not used in the class, though I also thought there was a check for that which seems not, so not sure

**Review by anuraaga — APPROVED** [review]

File: sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/state/WriteableMetricStorage.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (jkwatson) [reply sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/state/WriteableMetricStorage.java:10]: seems fine to me. Why does it matter if it's a link or not?

[the remark this answers is not in the document]
Comment (anuraaga) [reply sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/state/WriteableMetricStorage.java:10]: Sometimes I go through import statements of a file to see what dependencies it has, and javadoc-only imports can be a bit confusing. Not a big deal at all, but not zero either

[the remark this answers is not in the document]
Comment (jkwatson) [reply sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/state/WriteableMetricStorage.java:10]: Heh. I don't think I ever look at imports...just noise that the IDE can manage for me. :)

**Pull request merged**